src/main/java/com/iheartradio/m3u8/ParseUtil.java

```diff
@@ -54,17 +54,20 @@ public static float parseFloat(String string, String tag) throws ParseException
 
         if (matcher.matches()) {
             String valueString = matcher.group(1);
+            if (valueString.length() % 2 != 0) {
+                throw ParseException.create(ParseExceptionType.INVALID_HEXADECIMAL_STRING, tag, hexString);
+            }
```

Comment:
`parseHexadecimal` doesn't know what it's parsing, so it shouldn't assume the length of the string is a multiple of 2. If you want to validate the IV (is the IV required to have an even length?), then that validation probably belongs in `PlaylistValidation`.

Reply:
Since each byte maps to 2 hexadecimal digits it has to be a multiple of 2 - unless you want to assume the last half byte is all zeroes.

Reply:
That makes sense, let's keep it like you implemented.